ts/features/itwallet/walletInstance/store/reducers/index.ts

```diff
@@ -1,21 +1,37 @@
 import * as O from "fp-ts/lib/Option";
-import { flow } from "fp-ts/lib/function";
+import { flow, pipe } from "fp-ts/lib/function";
 import { PersistConfig, persistReducer } from "redux-persist";
 import { createSelector } from "reselect";
 import { getType } from "typesafe-actions";
 import { Action } from "../../../../../store/actions/types";
 import { GlobalState } from "../../../../../store/reducers/types";
 import itwCreateSecureStorage from "../../../common/store/storages/itwSecureStorage";
-import { isWalletInstanceAttestationValid } from "../../../common/utils/itwAttestationUtils";
+import {
+  isWalletInstanceAttestationValid,
+  toRevocationReason
+} from "../../../common/utils/itwAttestationUtils";
 import { itwLifecycleStoresReset } from "../../../lifecycle/store/actions";
-import { itwWalletInstanceAttestationStore } from "../actions";
+import {
+  itwWalletInstanceAttestationStore,
+  itwUpdateWalletInstanceStatus
+} from "../actions";
+
+export enum RevocationReason {
+  CERTIFICATE_REVOKED_BY_ISSUER = "CERTIFICATE_REVOKED_BY_ISSUER",
+  NEW_WALLET_INSTANCE_CREATED = "NEW_WALLET_INSTANCE_CREATED",
+  REVOKED_BY_USER = "REVOKED_BY_USER"
+}
```

Comment:
This can be extracted from `WalletInstanceStatus` (see above) with

```typescript
 type WalletInstanceRevocationReason =
  WalletInstanceStatus["revocation_reason"];
  ```